src/putty2john.c

```diff
@@ -271,6 +199,10 @@ static int init_LAME(const Filename *filename) {
 		strcpy(alg, "ssh-rsa");
 	else if (!strcmp(b, "ssh-dss"))
 		strcpy(alg, "ssh-dss");
+	else if (!strcmp(b, "ecdsa-sha2-nistp256"))
+		strcpy(alg, "ecdsa-sha2-nistp256");
+	else if (strlen(b) < sizeof(alg) - 1)
+		strcpy(alg, b);
```

Comment:
The last catch-all could actually replace lines 198-203, right? You could do with just
```c
	if (strlen(b) < sizeof(alg) - 1)
		strcpy(alg, b);
```